vision_layer/vision_tasks/include/buoy.h

```diff
@@ -1,53 +1,15 @@
 #ifndef BUOY_TASK_H
 #define BUOY_TASK_H
 
-#include "ros/ros.h"
-#include "sensor_msgs/Image.h"
-#include "opencv2/imgproc/imgproc.hpp"
-#include "opencv2/core/core.hpp"
-#include "opencv2/imgproc/imgproc_c.h"
-#include "opencv2/highgui/highgui.hpp"
-#include <cv_bridge/cv_bridge.h>
-#include <image_transport/image_transport.h>
-#include <dynamic_reconfigure/server.h>
-#include <geometry_msgs/PointStamped.h>
-#include <sensor_msgs/image_encodings.h>
-#include <bits/stdc++.h>
-#include <stdlib.h>
-#include <std_msgs/Bool.h>
-#include <std_msgs/Float32.h>
-#include <string>
-#include <boost/thread.hpp> 
-
+#include "base_class.h" 
 #include <vision_tasks/buoyRangeConfig.h>
-#include <filter.h>
-#include <contour.h>
-#include <morph.h>
-#include <threshold.h>
-
-#define SHELLSCRIPT_DUMP_RED_BUOY "\
-#/bin/bash \n\
-echo -e \"Parameters dumped!!\" \n\
-rosparam dump -v ~/Projects/anahita/src/Anahita/vision_layer/vision_tasks/thresholding/red_buoy.yaml /vision_node\
-"
-#define SHELLSCRIPT_LOAD_RED_BUOY "\
-#/bin/bash \n\
-echo -e \"Parameters loaded!!\" \n\
-rosparam load -v ~/Projects/anahita/src/Anahita/vision_layer/vision_tasks/thresholding/red_buoy.yaml /vision_node\
-"
 
-class Buoy
+class Buoy : public Base_class
```

Comment:
A very good coding practice is to create your own namespaces. I would strongly recommend doing that. Reason: [link](https://stackoverflow.com/questions/4211827/why-and-how-should-i-use-namespaces-in-c)